### PR TITLE
Category Filter

### PR DESCRIPTION
**Pull request by AymenBida**
In this milestone I did: 

- Add a category filter to the app which adds the option to choose which books the user can see depending on their respective categories

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @AymenBida  👋

**You've done pretty well here. The filter action and reducer were implemented correctly. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/components/App.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import BooksList from '../containers/BooksList';
 import BooksForm from '../containers/BooksForm';
+import CategoryFilter from './CategoryFilter';
 
 const App = () => (
   <div>
+    <CategoryFilter />
```
Comment (mikenath223) [inline src/components/App.js]: - [x] The CategoryFilter component should be imported into BookList container with the handleFilterChange function passed down to it. It shouldn't be placed in App.js.

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const CategoryFilter = () => {
+  const [category, setCategory] = useState('0');
+  const dispatch = useDispatch();
+
+  const handleChange = (event) => {
+    const categoryName = allCategories.find((cat) => cat.id === event.target.value).name;
+    setCategory(event.target.value);
+    dispatch(changeFilter(categoryName));
+  };
```
Comment (mikenath223) [inline src/components/CategoryFilter.js]: - [x] The redux store already stores the category therefore please remove this react state there is no need for it.

File: src/components/CategoryFilter.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <label htmlFor="filter">
+      Filter by Category
+      <select name="filter" id="filter" value={category} onChange={handleChange}>
+        {allCategories.map(
+          (cat) => <option key={cat.id} value={cat.id}>{cat.name}</option>,
+        )}
+      </select>
```
Comment (mikenath223) [inline src/components/CategoryFilter.js]: - [x] Simply read from the redux store the value of category.

**Review by bellom — APPROVED** [review]
## Hi @AymenBida  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request closed without merging**